client/quantum_serverless/core/files.py

```diff
@@ -0,0 +1,92 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""
+===============================================
+Provider (:mod:`quantum_serverless.core.files`)
+==============================================
+
+.. currentmodule:: quantum_serverless.core.files
+
+Quantum serverless files
+========================
+
+.. autosummary::
+    :toctree: ../stubs/
+
+"""
+import os.path
+import uuid
+from typing import List, Optional
+
+import requests
+from opentelemetry import trace
+from tqdm import tqdm
+
+from quantum_serverless.core.constants import REQUESTS_TIMEOUT
+from quantum_serverless.utils.json import safe_json_request
+
+
+class GatewayFilesClient:
+    """GatewayFilesClient."""
+
+    def __init__(self, host: str, token: str, version: str):
+        """Files client for Gateway service.
+
+        Args:
+            host: gateway host
+            version: gateway version
+            token: authorization token
+        """
+        self.host = host
+        self.version = version
+        self._token = token
+
+    def download(self, file: str, download_location: str) -> Optional[str]:
+        """Downloads file."""
+        tracer = trace.get_tracer("client.tracer")
+        with tracer.start_as_current_span("files.download"):
+            with requests.get(
+                f"{self.host}/api/{self.version}/files/download/",
+                params={"file": file},
+                stream=True,
+                headers={"Authorization": f"Bearer {self._token}"},
+                timeout=REQUESTS_TIMEOUT,
+            ) as req:
+                req.raise_for_status()
+
+                total_size_in_bytes = int(req.headers.get("content-length", 0))
+                chunk_size = 8192
+                progress_bar = tqdm(
+                    total=total_size_in_bytes, unit="iB", unit_scale=True
+                )
+                file_name = f"downloaded_{str(uuid.uuid4())[:8]}_{file}"
```

Comment:
Why does this download into an unique file name?  Avoiding override or appending?

Reply:
yes, to avoid overriding and we also return file name too, if user want to do something with it programmatically